### PR TITLE
Add pipeline testing with singularity

### LINKED ISSUE #10: Pipeline testing with singularity containers for snakemake rules
**Is your feature request related to a problem? Please describe.**
For the development time we might work in *conda* virtual environments but for the final version the *snakemake* pipeline should be executable with both `--use-conda` and `--use-singularity` flags. At some point CI needs to be testing the containerized execution as well.

**Describe the solution you'd like**
Add the call with `--use-singularity` to the CI; conda env. would have to be updated. Is it possible to use singularity on Travis at all?

### PR DESCRIPTION
**Pull request by krish8484**
## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #10  

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have not reduced the existing code coverage

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -16,6 +16,27 @@ install:
...
+  # Install singularity
+  - if [ "$TRAVIS_OS_NAME" = "linux" ]; then sudo apt-get update; fi
+  - if [ "$TRAVIS_OS_NAME" = "linux" ]; then sudo apt-get install -y build-essential; fi
+  - if [ "$TRAVIS_OS_NAME" = "linux" ]; then sudo apt-get install -y uuid-dev ; fi
+  - if [ "$TRAVIS_OS_NAME" = "linux" ]; then sudo apt-get install -y squashfs-tools; fi
+  - if [ "$TRAVIS_OS_NAME" = "linux" ]; then sudo apt-get install -y libseccomp-dev; fi
+  - if [ "$TRAVIS_OS_NAME" = "linux" ]; then sudo apt-get install -y wget; fi
+  - if [ "$TRAVIS_OS_NAME" = "linux" ]; then sudo apt-get install -y pkg-config; fi
+  - if [ "$TRAVIS_OS_NAME" = "linux" ]; then sudo apt-get install -y git; fi
+  - if [ "$TRAVIS_OS_NAME" = "linux" ]; then sudo apt-get install -y cryptsetup-bin; fi
```
[the remark this answers is not in the document]
Comment (krish8484) [reply .travis.yml]: Removed `update` and `git`. `wget` is necessary, see https://travis-ci.com/github/zavolanlab/binding-scanner/jobs/377011832

File: .travis.yml [hunk 2]
```diff
@@ -98,4 +119,4 @@ script:
   # Test pipeline execution: local, conda envs:
   - bash tests/integration/execution/snakemake_local_run_conda_environments.sh
   - md5sum --check tests/integration/expected_output.md5
-  
+  - if [ "$TRAVIS_OS_NAME" = "linux" ]; then bash tests/integration/execution/snakemake_local_run_singularity_environments.sh; fi
```
[the remark this answers is not in the document]
Comment (krish8484) [reply .travis.yml]: I have added the required commands. For `sequence_logos`, the `png` files are similar but the md5sums are not the same as `conda`. Should I add another file with separate checksums for singularity?

File: .travis.yml [hunk 3]
```diff
@@ -16,6 +16,27 @@ install:
...
+  - if [ "$TRAVIS_OS_NAME" = "linux" ]; then go version; fi
+  - if [ "$TRAVIS_OS_NAME" = "linux" ]; then export VERSION=3.5.2; fi # adjust this as necessary
+  - if [ "$TRAVIS_OS_NAME" = "linux" ]; then wget https://github.com/sylabs/singularity/releases/download/v${VERSION}/singularity-${VERSION}.tar.gz; fi
+  - if [ "$TRAVIS_OS_NAME" = "linux" ]; then tar -xzf singularity-${VERSION}.tar.gz; fi
+  - if [ "$TRAVIS_OS_NAME" = "linux" ]; then cd singularity; fi
+  - if [ "$TRAVIS_OS_NAME" = "linux" ]; then ./mconfig; fi
+  - if [ "$TRAVIS_OS_NAME" = "linux" ]; then make -C ./builddir; fi
+  - if [ "$TRAVIS_OS_NAME" = "linux" ]; then sudo make -C ./builddir install; fi
+  - if [ "$TRAVIS_OS_NAME" = "linux" ]; then singularity --version; fi
+  - if [ "$TRAVIS_OS_NAME" = "linux" ]; then cd ..; fi
```
[the remark this answers is not in the document]
Comment (krish8484) [reply .travis.yml:37]: Yes, there are multiple relative paths in `mconfig` and `builddir`. Running from the parent directory messes up these relative paths resulting in the unsuccessful installation of `singularity`.

File: .travis.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (AngryMaciek) [reply .travis.yml]: Yes, could you please do that?  
Then I guess there will be two versions of `expected_files.md5` for the integration test.

[the remark this answers is not in the document]
Comment (AngryMaciek) [reply .travis.yml]: I wonder why they are different, though... Can you notice any difference while running integration test locally with `conda` or `singularity`?

[the remark this answers is not in the document]
Comment (krish8484) [reply .travis.yml]: Even I don't understand why are they different, the versions are the same in `conda` and `singularity`. There is no noticeable difference in the image though.

**Pull request merged**